src/main/scala/com/microsoft/partnercatalyst/fortis/spark/analyzer/ExtendedFortisEvent.scala

```diff
@@ -0,0 +1,21 @@
+package com.microsoft.partnercatalyst.fortis.spark.analyzer
+
+import java.util.UUID
+
+import com.microsoft.partnercatalyst.fortis.spark.dto.{Analysis, Details, FortisEvent, Location}
+
+case class ExtendedFortisEvent[T](
+  details: ExtendedDetails[T],
+  analysis: Analysis
+) extends FortisEvent
+
+case class ExtendedDetails[T](
+  id: UUID,
+  createdAtEpoch: Long,
+  body: String,
+  title: String,
+  publisher: String,
+  sourceUrl: String,
+  sharedLocations: List[Location] = List(),
+  original: T
```

Comment:
Don't get me wrong; this approach is likely better than what there was before. I just question whether now was the right time to make these changes as opposed to just implementing what we needed, getting on with feature work and cleaning things up later if required. Given that the change is here, it's fine. That's why I signed off. It's just that this is the second huge pull request in two weeks, so going forward I'm wondering if we can implement functionality with fewer, smaller and more surgical changes.

I disagree that we wasted time with the initial approach as getting something up and running on which to iterate is always worth something. Now it's just about rearranging the blocks :)